### ISSUE TITLE
How can I build nmap with epoll?

### ISSUE THREAD

**Issue by fiht** [nmap/nmap#948]
# Question
I configured nmap from source and it told me I'm compiled without epoll.
```
➜  fiht_code nmap --version
Nmap version 7.50 ( https://nmap.org )
Platform: x86_64-unknown-linux-gnu
Compiled with: nmap-liblua-5.3.3 openssl-1.0.2g libpcre-8.38 libpcap-1.7.4 nmap-libdnet-1.12 ipv6
Compiled without:
Available nsock engines: epoll poll select
```
Could somebody tell me how can I compile nmap with epool?

MyOS: Ubuntu 16.04
Kernel: Linux ubuntu 4.12.0-041200-generic

# Why 
I'm using nmap scan a lot of ips. Because nmap perform a slow speed when testing a single IP. So I start 100 nmap progress in my server. Which is much more quick but it spend a lot of CPU. And I use nmap -v get the following output:
```
➜  fiht_code nmap --version

Nmap version 7.50 ( https://nmap.org )
Platform: x86_64-unknown-linux-gnu
Compiled with: nmap-liblua-5.3.3 openssl-1.0.2g libpcre-8.38 libpcap-1.7.4 nmap-libdnet-1.12 ipv6
Compiled without:
Available nsock engines: epoll poll select
```
it shows I'm Compiled without epoll. And I'm thinking Nmap could be  more efficiency by useing epoll.  I've googled but seems no one focus on that. Could someone give me a help?

Thanks!

**Comment by dmiller-nmap** [comment]
Actually, you do have epoll support built in. The output could be
confusing, but it means, "Compiled without: <empty list>. Compiled with
these Nsock engines: epoll, poll, select"

Dan

On Jul 23, 2017 8:45 AM, "方大核桃" <notifications@github.com> wrote:

Question

I configured nmap from source and it told me I'm compiled without epoll.

➜  fiht_code nmap --version
Nmap version 7.50 ( https://nmap.org )
Platform: x86_64-unknown-linux-gnu
Compiled with: nmap-liblua-5.3.3 openssl-1.0.2g libpcre-8.38
libpcap-1.7.4 nmap-libdnet-1.12 ipv6
Compiled without:
Available nsock engines: epoll poll select

Could somebody tell me how can I compile nmap with epool?

MyOS: Ubuntu 16.04
Kernel: Linux ubuntu 4.12.0-041200-generic
Why

I'm using nmap scan a lot of ips. Because nmap perform a slow speed when
testing a single IP. So I start 100 nmap progress in my server. Which is
much more quick but it spend a lot of CPU. And I use nmap -v get the
following output:

➜  fiht_code nmap --version

Nmap version 7.50 ( https://nmap.org )
Platform: x86_64-unknown-linux-gnu
Compiled with: nmap-liblua-5.3.3 openssl-1.0.2g libpcre-8.38
libpcap-1.7.4 nmap-libdnet-1.12 ipv6
Compiled without:
Available nsock engines: epoll poll select

it shows I'm Compiled without epoll. And I'm thinking Nmap could be more
efficiency by useing epoll. I've googled but seems no one focus on that.
Could someone give me a help?

Thanks!

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/nmap/nmap/issues/948>, or mute the thread
